src/pydata_sphinx_theme/utils.py

```diff
@@ -58,3 +60,102 @@ def maybe_warn(app: Sphinx, msg, *args, **kwargs):
     should_warn = theme_options.get("surface_warnings", False)
     if should_warn:
         SPHINX_LOGGER.warning(msg, *args, **kwargs)
+
+
+def set_secondary_sidebar_items(
+    app: Sphinx, pagename: str, templatename: str, context, doctree
+) -> None:
+    """Set the secondary sidebar items to render for the given pagename."""
+    if "theme_secondary_sidebar_items" in context:
+        templates = context["theme_secondary_sidebar_items"]
+        if isinstance(templates, dict):
+            templates = _get_matching_sidebar_items(pagename, templates)
+
+        context["secondary_sidebar_items"] = _update_and_remove_templates(
+            app,
+            context,
+            templates,
+            "theme_secondary_sidebar_items",
+        )
+
+
+def _update_and_remove_templates(
+    app: Sphinx,
+    context: Dict[str, Any],
+    templates: Union[List, str],
+    section: str,
+    templates_skip_empty_check: Optional[List[str]] = None,
+) -> List[str]:
+    """Update templates to include html suffix if needed; remove templates which render empty.
+
+    Args:
+        app: Sphinx application passed to the html page context
+        context: The html page context; dictionary of values passed to the templating engine
+        templates: A list of template names, or a string of comma separated template names
+        section: Name of the template section where the templates are to be rendered
+        templates_skip_empty_check: Names of any templates which should never be removed from the list
+            of filtered templates returned by this function. These templates aren't checked if they
+            render empty, which can save time if the template is slow to render.
+
+    Returns:
+        A list of template names (including '.html' suffix) to render into the section
+    """
+    if templates_skip_empty_check is None:
+        templates_skip_empty_check = []
+
+    # Break apart `,` separated strings so we can use , in the defaults
+    if isinstance(templates, str):
+        templates = [template.strip() for template in templates.split(",")]
+
+    # Add `.html` to templates with no suffix
+    suffixed_templates = []
+    for template in templates:
+        if os.path.splitext(template)[1]:
+            suffixed_templates.append(template)
+        else:
+            suffixed_templates.append(f"{template}.html")
+
+    ctx = copy.copy(context)
+    ctx.update({section: suffixed_templates})
```

Comment:
I'd recommend a quick check like `section in ["theme_secondary_sidebar_items", "theme_sidebar_items"]` (not sure I remembered the variables properly). These names are a bit hard to remember and this will make it clearer what this is meant for.

Reply:
Hmm, this function is called for [a bunch of different templates](https://github.com/peytondmurray/pydata-sphinx-theme/blob/bebc73bb011eaeb43147833e6b692b4a0d5d210f/src/pydata_sphinx_theme/__init__.py#L210), not just for the sidebar. I'm happy to put in a check here for the correct section name, though. Would it be best to get a list of all valid section names and make sure the section passed in is one of these allowed names?

```python
    template_sections = [
        "theme_navbar_start",
        "theme_navbar_center",
        "theme_navbar_persistent",
        "theme_navbar_end",
        "theme_article_header_start",
        "theme_article_header_end",
        "theme_article_footer_items",
        "theme_content_footer_items",
        "theme_footer_start",
        "theme_footer_center",
        "theme_footer_end",
        "theme_primary_sidebar_end",
        "theme_secondary_sidebar_items",
        "sidebars",
    ]

```

Reply:
Ahh I see - maybe just a comment in the docstring then to provide one or two examples of what the section names look like for reference, and link to some source of truth for all the section names